Sources/Bag.swift

```diff
@@ -22,11 +41,22 @@ public struct Bag<Element> {
 	/// - parameters:
 	///   - value: A value that will be inserted.
 	@discardableResult
-	public mutating func insert(_ value: Element) -> RemovalToken {
-		let token = RemovalToken()
-		let element = BagElement(value: value, token: token)
+	public mutating func insert(_ value: Element) -> Token {
+		var timestamp: UInt64 = 0
+
+		repeat {
+			// `DispatchTime` is expected to use monotonic clocks. That said a guard is
+			// put here just in case.
+			timestamp = DispatchTime.now().uptimeNanoseconds
+		} while timestamp <= lastTimestamp
+
+		lastTimestamp = timestamp
+
+		let token = Token(timestamp)
```

Comment:
At this point, it doesn't seem like there's any point in using a timestamp. This could just use a counter? `Bag` requires mutual exclusion for cross-thread use anyway.